### PR TITLE
EODP3-60: Officers List - Officer Quickview

### PR DESCRIPTION
**Pull request by tamilau**

### CODE REVIEW

File: src/aura/detailer_SailorQuickview/detailer_SailorQuickview.cmp [hunk 1]
```diff
@@ -0,0 +1,62 @@
+<aura:component description="detailer_SailorQuickview" controller="SailorQuickviewController">
+    <aura:attribute name="contactId" type="String" />
+    <aura:attribute name="contact" type="Object" />
+
+    <aura:handler name="change" value="{!v.contactId}" action="{!c.onContactChange}" />
+
+    <aura:registerEvent name="closeQuickview" type="c:componentEvent" />
```
Comment (P0und-cake) [inline src/aura/detailer_SailorQuickview/detailer_SailorQuickview.cmp]: @tamilau I know we spoke about this and didn't see much reason to create new events but I have one thing I want you to consider...if we have all our components handling the same event, we need to be mindful of unintentional consequences and perhaps even performance. I just worry that we'll lose track of what components are subscribed to this event and run into some not so obvious issues to debug later on.

Having said all that, do you have any concerns for this particular page?

File: src/classes/SailorQuickviewController.cls [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            else {
+                Contact contactRecord = contacts[0];
+                return new sailorQuickviewModel(contactRecord);
+            }
+        } catch (Exception e) {
+            throw new AuraHandledException(e.getMessage());
+        }
+    }
+
+    private static String buildStringDate(Date dateValue, String stringFormat) {
```
Comment (P0und-cake) [inline src/classes/SailorQuickviewController.cls]: I think its time we created a util for this sort of thing. What do you think about DisplayFormatUtil.cls? We could put this in there as well the name formatting?

File: src/aura/detailer_SailorQuickview/detailer_SailorQuickview.cmp (hunk 1 above)
Comment (tamilau) [reply src/aura/detailer_SailorQuickview/detailer_SailorQuickview.cmp]: @P0und-cake I couldn't find any reference to issues (and one blog post that advocated for this approach of one shared component). Good to keep in mind, and no concerns for this particular page.

File: src/classes/SailorQuickviewController.cls (hunk 2 above)
Comment (tamilau) [reply src/classes/SailorQuickviewController.cls]: Makes sense, I've added one.

**Pull request merged**